lib/Driver/ToolChains.cpp

```diff
@@ -142,6 +142,14 @@ static void addLTOArgs(const OutputInfo &OI, ArgStringList &arguments) {
   }
 }
 
+static void addLinkedLibArgs(const ArgList &Args, ArgStringList &FrontendArgs) {
+  for(auto Arg:Args.getAllArgValues(options::OPT_l))
+  {
+    std::string* lArg = new std::string("-l" + Arg);
+    FrontendArgs.push_back(lArg->c_str());
+  }
```

Comment:
1. Using `new` here will create a memory leak, since that memory is not owned by anything; notice that `ArgStringList` is a type alias for a vector, so storing pointers in the vector doesn't maintain any ownership.  Instead, you probably want to use `ArgList::MakeArgString`, so that the memory is owned by the `Args`.
2. Secondly, you probably want to call `Arg->claim()`. If you look at the body of `AddAllArgs` (which was being used earlier for `-l`) in LLVM's `ArgList.cpp`, you'll see that it calls `Arg->claim()` before calling `Arg->render`.

Nit: Please [format the code using `clang-format`](https://github.com/apple/swift/blob/main/docs/HowToGuides/FAQ.md#how-do-i-format-my-changes) or manually to match the existing style